### PR TITLE
Adding a hook for scene becoming active

### PR DESCRIPTION
**Pull request by pranavrajgopal**

### CODE REVIEW

File: start/AppDistributionExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,10 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		341261F8CC14E317A6E3B1CF /* Pods_AppDistributionExample_AppDistributionExampleUITests.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 630F229983A3E80E41E1F17B /* Pods_AppDistributionExample_AppDistributionExampleUITests.framework */; };
```
Comment (lfkellogg) [inline start/AppDistributionExample.xcodeproj/project.pbxproj:10]: We don't need any of the changes in this file, right?

File: start/AppDistributionExample/AppDelegate.swift [hunk 2]
```diff
@@ -13,13 +13,14 @@
 // limitations under the License.
 
 import UIKit
+import Firebase
```
Comment (lfkellogg) [inline start/AppDistributionExample/AppDelegate.swift:16]: Remove the changes in this file

File: start/AppDistributionExample/AppDistributionViewController.swift [hunk 3]
```diff
@@ -50,12 +51,41 @@ class AppDistributionViewController: UIViewController {
...
+    
+    func sceneBecameActive() {
+        checkForUpdate()
+    }
 
   override func viewDidAppear(_ animated: Bool) {
   }
 
   // MARK: - Firebase 🔥
   private func checkForUpdate() {
```
Comment (lfkellogg) [inline start/AppDistributionExample/AppDistributionViewController.swift:63]: Remove the changes to this method.

File: start/Podfile [hunk 4]
```diff
@@ -4,7 +4,7 @@
 target 'AppDistributionExample' do
   # Comment the next line if you don't want to use dynamic frameworks
   use_frameworks!
-
+  pod 'Firebase/AppDistribution'
```
Comment (lfkellogg) [inline start/Podfile:7]: Remove this

File: start/Podfile.lock [hunk 5]
```diff
@@ -1,23 +1,9 @@
 PODS:
```
Comment (lfkellogg) [inline start/Podfile.lock:1]: Remove changes to this file

File: start/AppDistributionExample/AppDistributionViewController.swift [hunk 6]
```diff
@@ -50,12 +51,41 @@ class AppDistributionViewController: UIViewController {
     configureSignInStatus()
     view.addSubview(signedInStatus!)
   }
+    
+    func sceneBecameActive() {
+        checkForUpdate()
```
Comment (lfkellogg) [inline start/AppDistributionExample/AppDistributionViewController.swift:56]: In the `start` version this method would be empty. Maybe there could be a comment like the one in SceneDelegate: `// Called when the scene has moved from an inactive state to an active state.`

File: start/AppDistributionExample/AppDistributionViewController.swift [hunk 7]
```diff
@@ -50,12 +51,41 @@ class AppDistributionViewController: UIViewController {
     configureSignInStatus()
     view.addSubview(signedInStatus!)
   }
+    
+    func sceneBecameActive() {
```
Comment (lfkellogg) [inline start/AppDistributionExample/AppDistributionViewController.swift:55]: This is indented too far

**Review by lfkellogg — CHANGES_REQUESTED** [review]
I am concerned that people will see the method and wonder why it's there if it's not used, or even accidentally paste code in it.

What if `appDistributionViewController?.sceneBecameActive()` was commented out, and in the Dory, we'd say the fix is to uncomment that and add the `sceneBecameActive()` snippet to the view controller?

If we're not comfortable updating the example app at all, we could just paste everything into the Q&A, like "update your SceneDelegate to this code: `<snippet>` and add this code to your view controller: `<snippet>`". Instead of pasting the full SceneDelegate snippet into the Q&A we could even link to the raw code on a github branch, for example. I think that would be my preference, actually.

Also for any code change you would need to update the `final` version of the project as well.

File: start/AppDistributionExample.xcodeproj/project.pbxproj (hunk 1 above)
Comment (pranavrajgopal) [reply start/AppDistributionExample.xcodeproj/project.pbxproj:10]: Nope. I don't know why these got added. We should get rid of these if we plan to merge this.

File: start/AppDistributionExample/AppDistributionViewController.swift (hunk 7 above)
Comment (pranavrajgopal) [reply start/AppDistributionExample/AppDistributionViewController.swift:55]: Yup. Will fix!

**Pull request closed without merging**